domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,289 @@
+const val TEST_EXPLORATION_ID_1 = "test_exp_id_1"
+private const val WELCOME_EXPLORATION_DATA_PROVIDER_ID = "WelcomeExplorationDataProvider"
+private const val ABBOUT_OPPIA_EXPLORATION_DATA_PROVIDER_ID = "AboutOppiaExplorationDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class ExplorationDataController @Inject constructor(private val context: Context,
+                                                    private val dataProviders: DataProviders
+) {
+
+  private val welcomeExplorationDataProvider =
+    dataProviders.createInMemoryDataProviderAsync(
+      WELCOME_EXPLORATION_DATA_PROVIDER_ID, this::retrieveWelcomeExplorationAsync)
+  private val aboutOppiaExplorationDataProvider =
+    dataProviders.createInMemoryDataProviderAsync(
+      ABBOUT_OPPIA_EXPLORATION_DATA_PROVIDER_ID, this::retrieveAbboutOppiaExplorationAsync)
+
+  /**
+   * Returns an  [Exploration] given an ID.
+   */
+  fun getExplorationById(ID: String): LiveData<AsyncResult<Exploration>>? {
+    if (ID == TEST_EXPLORATION_ID_0) {
+      return dataProviders.convertToLiveData(welcomeExplorationDataProvider)
+    }
+    if (ID == TEST_EXPLORATION_ID_1) {
+      return dataProviders.convertToLiveData(aboutOppiaExplorationDataProvider)
+    }
+    return null
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveWelcomeExplorationAsync(): AsyncResult<Exploration> {
+    return AsyncResult.success(createExploration("welcome.json"))
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveAbboutOppiaExplorationAsync(): AsyncResult<Exploration> {
+    return AsyncResult.success(createExploration("about_oppia.json"))
+  }
+
+  // Returns an exploration given an assetName
+  private fun createExploration(assetName: String): Exploration {
+    val explorationObject = loadJSONFromAsset(assetName)?: return Exploration.getDefaultInstance()
+    return Exploration.newBuilder()
+      .setTitle(explorationObject?.getString("title"))
+      .setLanguageCode(explorationObject?.getString("language_code"))
+      .setInitStateName(explorationObject?.getString("init_state_name"))
+      .setObjective(explorationObject?.getString("objective"))
+      .putAllStates(createStatesFromJsonObject(explorationObject?.getJSONObject("states")))
+      .build()
+  }
+
+  // Returns a JSON Object if it exists, else returns null
+  private fun getJSONObject(parentObject: JSONObject, key: String): JSONObject? {
+    try {
+      return parentObject.getJSONObject(key)
+    } catch(e: JSONException) {
+      return null
+    }
+  }
+
+  // Loads the JSON string from an asset and converts it to a JSONObject
+  private fun loadJSONFromAsset(assetName: String): JSONObject? {
+    val am = context.assets
+
+    var jsonObject: JSONObject?
+    try {
+      val `is` = am.open(assetName)
```

Comment:
Fortunately, I believe we can simplify this with Kotlin I/O extensions (see: https://stackoverflow.com/a/39500046/3689782):

```kotlin
val assetManager = context.assets
val jsonContents = assetManager.open(assetName).bufferedReader().use { it.readText() }
return JSONObject(jsonContents)
```

Also: what does `` `is` `` do? I haven't seen that before.

Reply:
Done. `is` was just to allow usage of the variable 'is'  (for input stream) because otherwise its a keyword